packages/bns/src/index.ts

```diff
@@ -0,0 +1,748 @@
+import {
+  bufferCV,
+  callReadOnlyFunction,
+  ClarityType,
+  ClarityValue,
+  cvToString,
+  getAddressFromPrivateKey,
+  hash160,
+  makeRandomPrivKey,
+  makeUnsignedContractCall,
+  privateKeyToString,
+  ResponseErrorCV,
+  StacksTransaction,
+  standardPrincipalCV,
+  UnsignedContractCallOptions,
+} from '@stacks/transactions';
+
+import { StacksNetwork } from '@stacks/network';
+
+import { bufferCVFromString, decodeFQN, getZonefileHash, uintCVFromBN } from './utils';
+
+import BN from 'bn.js';
+
+export const BNS_CONTRACT_ADDRESS = 'ST000000000000000000002AMW42H';
+export const BNS_CONTRACT_NAME = 'bns';
+
+export interface PriceFunction {
+  base: BN;
+  coefficient: BN;
+  b1: BN;
+  b2: BN;
+  b3: BN;
+  b4: BN;
+  b5: BN;
+  b6: BN;
+  b7: BN;
+  b8: BN;
+  b9: BN;
+  b10: BN;
+  b11: BN;
+  b12: BN;
+  b13: BN;
+  b14: BN;
+  b15: BN;
+  b16: BN;
+  nonAlphaDiscount: BN;
+  noVowelDiscount: BN;
+}
+
+export interface BnsContractCallOptions {
+  functionName: string;
+  functionArgs: ClarityValue[];
+  publicKey: string;
+  network: StacksNetwork;
+  attachment?: Buffer;
+}
+
+async function makeBnsContractCall(options: BnsContractCallOptions): Promise<StacksTransaction> {
+  const txOptions: UnsignedContractCallOptions = {
+    contractAddress: BNS_CONTRACT_ADDRESS,
```

Comment:
Then I suppose this would be something like
```ts
contractAddress: options.network.chainId === ChainID.Mainnet ? BnsContractAddress.mainnet ? BnsContractAddress.testnet
```
?